cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/replication/ReplicationPolicyManagerTest.java

```diff
@@ -0,0 +1,349 @@
+        verify(mockReplicator2, never()).start();
+        verify(mockReplicator3, never()).start();
+        rpm.start();
+        rpm.stop();
+        rpm.start();
+        verify(mockReplicator1, times(2)).start();
+        verify(mockReplicator2, times(2)).start();
+        verify(mockReplicator3, times(2)).start();
+    }
+
+    /**
+     * Check that when the {@link }ReplicationPolicyManager} is setup with a single
+     * {@link Replicator}, when {@link ReplicationPolicyManager#start()} is called for a
+     * second time after the replicator completed normally, the {@link Replicator#start()} method
+     * is called a second time on the {@link Replicator}.
+     */
+    @Test
+    public void testRestartCompletedSingleReplicator() {
+        Replicator mockReplicator = mock(Replicator.class);
+        EventBus mockEventBus = new EventBus();
+        when(mockReplicator.getEventBus()).thenReturn(mockEventBus);
+        ReplicationPolicyManager rpm = new TestPolicy();
+        rpm.addReplicators(mockReplicator);
+        verify(mockReplicator, never()).start();
+        rpm.start();
+        mockEventBus.post(new ReplicationCompleted(mockReplicator, 1, 1));
+        rpm.start();
+        verify(mockReplicator, times(2)).start();
+    }
+
+    /**
+     * Check that when the {@link }ReplicationPolicyManager} is setup with multiple
+     * {@link Replicator}s, when {@link ReplicationPolicyManager#start()} is called for a
+     * second time after each replicator completed normally, the {@link Replicator#start()} method
+     * is called a second time on the {@link Replicator}s.
+     */
+    @Test
+    public void testRestartCompletedMultipleReplicators() {
+        Replicator mockReplicator1 = mock(Replicator.class);
+        Replicator mockReplicator2 = mock(Replicator.class);
+        Replicator mockReplicator3 = mock(Replicator.class);
+        EventBus mockEventBus1 = new EventBus();
+        EventBus mockEventBus2 = new EventBus();
+        EventBus mockEventBus3 = new EventBus();
+        when(mockReplicator1.getEventBus()).thenReturn(mockEventBus1);
+        when(mockReplicator2.getEventBus()).thenReturn(mockEventBus2);
+        when(mockReplicator3.getEventBus()).thenReturn(mockEventBus3);
+        ReplicationPolicyManager rpm = new TestPolicy();
+        rpm.addReplicators(mockReplicator1, mockReplicator2, mockReplicator3);
+        verify(mockReplicator1, never()).start();
+        verify(mockReplicator2, never()).start();
+        verify(mockReplicator3, never()).start();
+        rpm.start();
+        mockEventBus1.post(new ReplicationCompleted(mockReplicator1, 1, 1));
+        mockEventBus2.post(new ReplicationCompleted(mockReplicator2, 1, 1));
+        mockEventBus3.post(new ReplicationCompleted(mockReplicator3, 1, 1));
+        rpm.start();
+        verify(mockReplicator1, times(2)).start();
+        verify(mockReplicator2, times(2)).start();
+        verify(mockReplicator3, times(2)).start();
```

Comment:
Perhaps add a test where if you don't emit the complete event for one of the replicators, that doesn't have start called again? Similarly for the error test below.